data-science-onramp/data-ingestion/setup-test.py

```diff
@@ -0,0 +1,210 @@
+
+    # Create cluster configuration
+    zone_uri = \
+        'https://www.googleapis.com/compute/v1/projects/{}/zones/{}'.format(
+            project, zone)
+    cluster_data = {
+        'project_id': project,
+        'cluster_name': cluster_name,
+        'config': {
+            'gce_cluster_config': {
+                'zone_uri': zone_uri,
+                "metadata": {
+                    "PIP_PACKAGES": "google-cloud-storage"
+                },
+            },
+            'master_config': {
+                'num_instances': 1,
+                'machine_type_uri': 'n1-standard-8'
+            },
+            'worker_config': {
+                'num_instances': 6,
+                'machine_type_uri': 'n1-standard-8'
+            },
+            "initialization_actions": [
+                {
+                    "executable_file": ("gs://dataproc-initialization-actions/"
+                                        "python/pip-install.sh"),
+                }
+            ],
+            "software_config": {
+                "image_version": "1.5.4-debian10",
+                "optional_components": [
+                    "ANACONDA"
+                ],
+            }
+        }
+    }
+
+    # Create cluster using cluster client
+    cluster_client = dataproc.ClusterControllerClient(client_options={
+        'api_endpoint': '{}-dataproc.googleapis.com:443'.format(region)
+    })
+
+    cluster = cluster_client.create_cluster(project, region, cluster_data)
+    cluster.add_done_callback(callback)
+
+    # Wait for cluster to provision
+    global waiting_cluster_callback
+    waiting_cluster_callback = True
+
+    wait_for_cluster_creation()
+
+    # Create job configuration
+    job_details = {
+        'placement': {
+            'cluster_name': cluster_name
+        },
+        'pyspark_job': {
+            'main_python_file_uri': job_file_name,
+            'args': [
+                bucket_name,
+                "--test",
+            ],
+            "jar_file_uris": [
+                "gs://spark-lib/bigquery/spark-bigquery-latest_2.12.jar"
+            ],
+        },
+    }
+
+    # Submit job to dataproc cluster
+    job_client = dataproc.JobControllerClient(client_options={
+        'api_endpoint': '{}-dataproc.googleapis.com:443'.format(region)
+    })
+
+    result = job_client.submit_job(project_id=project, region=region,
+                                   job=job_details)
+
+    job_id = result.reference.job_id
+    print('Submitted job \"{}\".'.format(job_id))
+
+    # Wait for job to complete
+    wait_for_job(job_client, job_id)
+
+    # Get job output
+    cluster_info = cluster_client.get_cluster(project, region, cluster_name)
+    bucket = storage_client.get_bucket(cluster_info.config.config_bucket)
+    output_blob = (
+        'google-cloud-dataproc-metainfo/{}/jobs/{}/driveroutput.000000000'
+        .format(cluster_info.cluster_uuid, job_id))
+    out = bucket.blob(output_blob).download_as_string().decode("utf-8")
```

Comment:
The way to do this has actually JUST been updated about a week ago to be a bit easier to work with. Apologies as I haven't had the chance to update the samples yet. You can use [this method](https://googleapis.dev/python/dataproc/latest/gapic/v1/api.html?highlight=jobcontrollerclient#google.cloud.dataproc_v1.JobControllerClient.submit_job_as_operation)

Something like this should work (might need tweaking):
```
operation = job_client.submit_job_as_operation(project_id=project, region=region,
                                   job=job_details)

# This will wait for the job to finish before continuing.
result = operation.result() 

output_location = result.driver_output_resource_uri + ".000000000"
output = bucket.blob(output_location).download_as_string().decode("utf-8")
````